BranchMetrics/android-branch-deep-linking-attribution
# Core 1588 hashed partner params

## Reference
https://branch.atlassian.net/browse/CORE-1588.

## Description
Tech spec https://docs.google.com/document/d/1IF6JN4NZenVdhTqSGwUe04yqp0GwmNpkoyOR5dHlLa8/edit#heading=h.16u1uxt2gltj
Android ticket is linked above but a little sparse, iOS ticket has a little more context https://branch.atlassian.net/browse/CORE-1521

Basically. I added two APIs
- addFacebookPartnerParameterWithName(@NonNull String key, @NonNull String value)
- clearPartnerParameters()
the APIs will attach the following blob to the top lelevel of the json response to all install/open and custom/standard event requests

```
  "partner_data": {
    "fb": {
      "em": "194b86d986ad041666822dad7602f1a7bac1d9e286273e86141666ffb4b1909b",
      "ph": "7cd2dfb0b893767d1b0a4cacd831f6f9cf0ace6b1fbfc616c313c672d08f6196"
    }
  }
```


## Testing Instructions
Run the testbed app, engage a bunch of APIs and ensure that only open/install/events contain the above blob. 

## Risk Assessment [`HIGH` || `MEDIUM` || `LOW`]
LOW


## Reviewer Checklist (To be checked off by the reviewer only)

- Correctness & Style
